apps/api/src/app/subscribers-v2/usecases/list-subscribers/list-subscribers.usecase.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@nestjs/common';
+import { InstrumentUsecase } from '@novu/application-generic';
+import { SubscriberRepository } from '@novu/dal';
+import { DirectionEnum, IListSubscribersResponseDto } from '@novu/shared';
+import { ListSubscribersCommand } from './list-subscribers.command';
+
+@Injectable()
+export class ListSubscribersUseCase {
+  constructor(private subscriberRepository: SubscriberRepository) {}
+
+  @InstrumentUsecase()
+  async execute(command: ListSubscribersCommand): Promise<IListSubscribersResponseDto> {
+    const query = {
+      _environmentId: command.user.environmentId,
+      _organizationId: command.user.organizationId,
+    } as const;
+
+    if (command.query || command.email || command.phone || command.subscriberId || command.name) {
+      const searchConditions: Record<string, unknown>[] = [];
+
+      if (command.query) {
+        searchConditions.push(
+          ...[
+            { subscriberId: { $regex: command.query, $options: 'i' } },
+            { email: { $regex: command.query, $options: 'i' } },
+            { phone: { $regex: command.query, $options: 'i' } },
+            {
+              $expr: {
+                $regexMatch: {
+                  input: { $concat: ['$firstName', ' ', '$lastName'] },
+                  regex: command.query,
+                  options: 'i',
+                },
+              },
+            },
+          ]
+        );
+      }
+
+      if (command.email) {
+        searchConditions.push({ email: { $regex: command.email, $options: 'i' } });
+      }
+
+      if (command.phone) {
+        searchConditions.push({ phone: { $regex: command.phone, $options: 'i' } });
+      }
+
+      if (command.subscriberId) {
+        searchConditions.push({ subscriberId: { $regex: command.subscriberId, $options: 'i' } });
+      }
+
+      if (command.name) {
+        searchConditions.push({
+          $expr: {
+            $regexMatch: {
+              input: { $concat: ['$firstName', ' ', '$lastName'] },
+              regex: command.name,
+              options: 'i',
+            },
+          },
+        });
+      }
+
+      Object.assign(query, { $or: searchConditions });
```

Comment:
Not sure about this, but shouldn't it be $and any time except the query params? That's how search behaves in activity feed today.

Reply:
You're right. It should be `$and` and I removed the query param as well.